src/neo/SmartContract/Native/NativeContract.cs

```diff
@@ -28,7 +28,7 @@ public abstract class NativeContract
         public uint ServiceHash { get; }
         public byte[] Script { get; }
         public UInt160 Hash { get; }
-        public Guid Guid { get; }
+        public abstract uint Id { get; }
```

Comment:
Can't we read the real id?

Reply:
Yes, `ContractId `cannot be read for native contracts, so it needs to be specified in the native contract

Reply:
Why not? It should be.

Reply:
Since it will affect the real `ContractId`, if we want to add more native contracts in future. Therefore, we want to isolate the native contract id from user contract id.

Reply:
Maybe native could be negative, and not native positive, we can use the same counter for both.

Reply:
Shargon's idea is not bad, because we could simply start with -1, -2 and keep incrementing as new native appears.

Reply:
> Maybe native could be negative, and not native positive, we can use the same counter for both.

Good idea.